### PR TITLE
Add helper function `is_vip_convenience_domain()`

### PR DESCRIPTION
**Pull request by rebeccahum**
## Description
Add helper function `is_vip_convenience_domain()` (with updates to codebase) and updates to https://github.com/Automattic/vip-go-mu-plugins/pull/5129 to use it

## Pre-review checklist

Please make sure the items below have been covered before requesting a review:

- [ ] This change works and has been tested locally (or has an appropriate fallback).
- [ ] This change works and has been tested on a Go sandbox.
- [x] This change has relevant unit tests (if applicable).
- [ ] This change uses a rollout method to ease with deployment (if applicable - especially for large scale actions that require writes).
- [ ] This change has relevant documentation additions / updates (if applicable).
- [x] I've created a changelog description that aligns with the provided examples.

## Pre-deploy checklist

- [ ] VIP staff: Ensure any alerts added/updated conform to internal standards (see internal documentation).

### CODE REVIEW

**Review by sjinks — APPROVED** [review]

File: 001-core/privacy.php [hunk 1]
```diff
@@ -465,11 +465,12 @@ function delete_old_export_files() {
  * @return string The modified robots.txt content.
  */
 function vip_convenience_domain_robots_txt( $output ) {
-	$host = strtolower( $_SERVER['HTTP_HOST'] ?? '' ); // phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
-	if ( false !== strpos( $host, '.go-vip.co' ) || false !== strpos( $host, '.go-vip.net' ) ) {
+	if ( is_vip_convenience_domain( strtolower( $_SERVER['HTTP_HOST'] ?? '' ) ) ) { // phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
```
Comment (GaryJones) [inline 001-core/privacy.php:468]: The `is_vip_convenience_domain()` function already calls `strtolower()`, so it's not needed here.

File: vip-helpers/vip-utils.php [hunk 2]
```diff
@@ -1640,3 +1640,19 @@ function vip_get_hyper_servers( $hyperdb, $operation = 'all', $dataset = 'global
...
 }
+
+/**
+ * Checks if a given domain is a VIP convenience domain. Examples of a VIP convenience
+ * domain are `example.go-vip.co` and `example.go-vip.net`.
+ *
+ * @param string $domain Domain to check
+ * @return bool True if the domain is a convenience domain
+ */
+function is_vip_convenience_domain( string $domain ) {
```
Comment (GaryJones) [inline vip-helpers/vip-utils.php]: Slightly surprised that these utility functions aren't already under a namespace. There's a tiny but non-zero chance of a namespace clash here.

Comment (GaryJones) [inline vip-helpers/vip-utils.php]:
```suggestion
function is_vip_convenience_domain( string $domain ): bool {
```

File: vip-helpers/vip-utils.php [hunk 3]
```diff
@@ -1640,3 +1640,19 @@ function vip_get_hyper_servers( $hyperdb, $operation = 'all', $dataset = 'global
 
 	return $hyperdb->hyper_servers[ $dataset ];
 }
+
+/**
+ * Checks if a given domain is a VIP convenience domain. Examples of a VIP convenience
+ * domain are `example.go-vip.co` and `example.go-vip.net`.
+ *
+ * @param string $domain Domain to check
+ * @return bool True if the domain is a convenience domain
```
Comment (GaryJones) [inline vip-helpers/vip-utils.php]:
```suggestion
 * Checks if a given string looks like a VIP convenience domain.
 *
 * Examples of a VIP convenience domain are `example.go-vip.co` and `example.go-vip.net`.
 *
 * @param string $domain Domain to check.
 * @return bool True if the domain is a convenience domain.
```

File: vip-helpers/vip-utils.php [hunk 4]
```diff
@@ -1640,3 +1640,19 @@ function vip_get_hyper_servers( $hyperdb, $operation = 'all', $dataset = 'global
...
+ */
+function is_vip_convenience_domain( string $domain ) {
+	if ( '' === $domain ) {
+		return false;
+	}
```
Comment (GaryJones) [inline vip-helpers/vip-utils.php]: What's the advantage of breaking this bit out?

File: 001-core/privacy.php [hunk 5]
```diff
@@ -465,11 +465,12 @@ function delete_old_export_files() {
...
 		$output  = "# Crawling is blocked for go-vip.co and go-vip.net domains\n";
 		$output .= "User-agent: *\n";
 		$output .= "Disallow: /\n";
+	
+		return $output;
```
Comment (GaryJones) [inline 001-core/privacy.php]: Why is this extra `return` needed now?

**Review by GaryJones — CHANGES_REQUESTED** [review]

File: vip-helpers/vip-utils.php (hunk 2 above)
Comment (rebeccahum) [reply vip-helpers/vip-utils.php]: I did an audit already and there shouldn't be any clashes.

File: vip-helpers/vip-utils.php (hunk 4 above)
Comment (rebeccahum) [reply vip-helpers/vip-utils.php]: faster by 0.000000001 seconds duh

File: 001-core/privacy.php (hunk 5 above)
Comment (rebeccahum) [reply 001-core/privacy.php]: ??? What extra `return`? There's only one in the function

Comment (GaryJones) [reply 001-core/privacy.php]: Are you sure?

Screenshot 2024-01-05 at 20 38 29

**Review by luiztiago — APPROVED** [review]

**Review by GaryJones — APPROVED** [review]

**Comment by rebeccahum** [comment]
Blocked until https://github.com/Automattic/vip-go-mu-plugins/pull/5132 is deployed

**Pull request merged**